### PR TITLE
Enhance index filters

### PR DESCRIPTION
**Pull request by DarioSacco**
Some new filters and enhancements in modules index page:
- "More filters" toggle
- filter by categories
- filter by position, with additional "descendants" switch to change between api filters `parent` and `ancestor`
- multiple status filter
- filters initial values

Other things:
- changed to `false` the default value of the "menu" toggle in view's `Position` section

this filters can be enabled by object properties configuration, setting the right values for the property `filter`.
E.g.

```php
'Properties' => [
  'events' => [
    'filter' => [
      'status',
      'lang',
      'categories'
      'parent'
    ]
  ]
]
```

### CODE REVIEW

File: src/Form/Options.php [hunk 1]
```diff
@@ -58,13 +58,14 @@ public static function customControl(string $name, $value): array
...
      * Options for lang, using configuration loaded from API
-     * If available, use config `Project.config.I18n.languages`
+     * If available, use config `Project.config.I18n.languages`.
      *
      * @param mixed|null $value The field value.
      * @return array
      */
     public static function lang($value): array
     {
+        // $languages = Configure::read('I18n.languages');
```
Comment (didoda) [inline src/Form/Options.php]: This line can be removed

File: src/Template/Element/Form/bulk_export.twig [hunk 2]
```diff
@@ -1,32 +1,29 @@
 {% if not currentModule.hints.multiple_types %}
-    {{ Form.create(null, {'id': 'form-export', 'class': 'bulk-actions', 'url': {'_name': 'export:export', 'object_type': objectType}})|raw }}
-
+    {{ Form.create(null, {
+        'id': 'form-export',
+        'url': {'_name': 'export:export', 'object_type': objectType}
+    })|raw }}
```
Comment (didoda) [inline src/Template/Element/Form/bulk_export.twig:5]: class `bulk-actions` is missing, compared to previous version. is it desired?

File: src/Template/Element/Form/bulk_trash.twig [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
-
-            </div>
-
-        </div>
-
-    {{ Form.end()|raw }}
-
+{{ Form.create(null, {
+    'id': 'form-delete',
+    'url': {'_name': 'modules:delete', 'object_type': objectType}
```
Comment (didoda) [inline src/Template/Element/Form/bulk_trash.twig:4]: class `bulk-actions` missing here too

File: src/Controller/Component/PropertiesComponent.php [hunk 4]
```diff
@@ -60,7 +60,7 @@ class PropertiesComponent extends Component
         ],
 
         'filter' => [
-            'status',
+            'status'
```
Comment (stefanorosanelli) [inline src/Controller/Component/PropertiesComponent.php]: Ending `,` is required by coding style

File: src/Template/Element/Form/bulk_export.twig (hunk 2 above)
Comment (DarioSacco) [reply src/Template/Element/Form/bulk_export.twig:5]: yes. after some refactor in the style, that class was unnecessary

File: src/Template/Element/Form/bulk_trash.twig (hunk 3 above)
Comment (DarioSacco) [reply src/Template/Element/Form/bulk_trash.twig:4]: same here, I leaved that class only in the actions container

**Review by stefanorosanelli — APPROVED** [review]

**Pull request merged**